app/main.py

```diff
@@ -1,3 +1,79 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, distance: Distance) -> None:
+        self.km = distance
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance or int) -> Distance:
+        if isinstance(other, Distance):
+            self.km = self.km + other.km
+        else:
+            self.km = self.km + other.real
+        return Distance(distance=self.km)
+
+    def __iadd__(self, other: Distance or int) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            # return Distance(distance=self.km + other.real)
+            self.km += other.real
+        # return Distance(distance=self.km)
+        return self
+
+    def __mul__(self, other: Distance or int) -> Distance or None:
+        if isinstance(other, Distance):
+            return None
+        else:
+            return Distance(distance=self.km * other.real)
+
+    def __truediv__(self, other: Distance or int) -> Distance or None:
+        if isinstance(other, Distance):
+            return None
+        else:
+            return Distance(distance=round(self.km / other.real, 2))
+
+    def __eq__(self, other: Distance or int) -> Distance:
+        if isinstance(other, Distance):
+            return self.km == other.km
+        else:
+            return self.km == other.real
```

Comment:
Inside of your annotations use logical operator `|`, it works like `or`. Also, all of this cases might have `float` type of km, you should add annotation to this too.

Refactor all functions yousing only 1 `return`. Check your `return` annotation, are you sure that you are returning `Distance` instance in all methods?